lib/CFRPageActions.jsm

```diff
@@ -68,17 +85,58 @@ class PageAction {
     this._showPopupOnClick = this._showPopupOnClick.bind(this);
     this.dispatchUserAction = this.dispatchUserAction.bind(this);
 
-    this._l10n = new DOMLocalization([
-      "browser/newtab/asrouter.ftl",
-      "browser/branding/brandings.ftl",
-      "browser/branding/sync-brand.ftl",
-      "branding/brand.ftl",
-    ]);
+    this._l10n = this._createDOML10n();
 
     // Saved timeout IDs for scheduled state changes, so they can be cancelled
     this.stateTransitionTimeoutIDs = [];
   }
 
+  /**
+   * Creates a new DOMLocalization instance with the Fluent file from Remote Settings.
+   * Note that it still uses the packaged Fluent file as the fallback if the remote one
+   * is not available.
+   */
+  _createDOML10n() {
+    async function* generateBundles(resourceIds) {
+      const appLocale = Services.locale.appLocaleAsBCP47;
+      const appLocales = Services.locale.appLocalesAsBCP47;
+      const l10nFluentDir = OS.Path.join(
+        OS.Constants.Path.localProfileDir,
+        RS_DOWNLOADED_FILE_SUBDIR
+      );
+      const fs = new FileSource("cfr", [appLocale], `file://${l10nFluentDir}/`);
```

Comment:
I'm surprised this doesn't include `RS_FLUENT_VERSION` somewhere in the filename path.

Reply:
Yeah, I think it makes sense to have one copy of attachment here. Since the attachments (Fluent files) do not know anything about the versions, rather, its version is completely decided by its corresponding record (e.g. `cfr-v1-en-US`). Upon locale or version changes, it will re-download the attachment accordingly and overwrite the old one.